app/api/assignmentApiSerivce.js

```diff
@@ -210,13 +206,11 @@ angular
          * @return post_id : id of the deleted post
          */
         assignmentApi.deletePost = function (id) {
-            var url = '/admin/delete_post.json',
-                data = { post_id: id };
+            var url = '/api/admin/posts/' + id;
 
             return $http({
-                method: 'POST',
+                method: 'DELETE',
```

Comment:
So this is what I was talking about that we don't actually DELETE the resource, we modify a flag of `deleted`.  Because of this, it is part of the PUT.  You'll want to update the post with `"deleted": true` within the json blob.